### DISCUSSION TITLE
case CONNECT not working

### DISCUSSION THREAD

**Discussion by KoYejune0302** [ANLAB-KAIST/KENSv3#199] [Project Q&A 2. Transport Layer (TCP)] [answered]
3-way-handshake를 만든뒤 부터 case CONNECT에 변경사항이 제대로 적용이 안됩니다. 다른 case에 std::cout으로 출력을 하면 전부 출력이 되는데 connect 안에서만 작동을 하지 않습니다. 또한, connect에서 sendPackage를 주석 처리하였는데도 packet이 전송이 됩니다. build 디렉토리를 삭제하고 다시 진행해도 동일한 결과가 나옵니다.
<img width="819" alt="image" src="https://github.com/ANLAB-KAIST/KENSv3/assets/57734106/e17696a5-c851-4f0a-a69f-fcbaa792cb14">
<img width="1582" alt="image" src="https://github.com/ANLAB-KAIST/KENSv3/assets/57734106/ebf96f60-9041-4534-99fe-6dc392e2ae0b">

**Comment by Pusnow** [comment, marked as answer]
It is normal situation, since we test your TCP implementation with our implementation. 

For test cases staring with `TestAccept_`, your server-side codes (accept(), etc.) are tested.
For test cases staring with `TestConnect_`, your client-side codes (connect(), etc.) are tested.

For your case, it is probably a `TestAccept_` test case, which test your server-side codes, and in the test case, only TA's `connect()` is called, so it is normal to not using your `connect()` codes.

**Reply by KoYejune0302** [reply]
그런데 packet을 보내는 코드가 없는데도 packet이 전송이 될 수가 있나요?

**Reply by Pusnow** [reply]
Yes, because `sendPacket()` for `connect()` is called in TA's TCP implementation.
